### DISCUSSION TITLE
Build system config file

### DISCUSSION THREAD

**Discussion by ygorpontelo** [c3lang/c3c#1309] [Ideas]
Hey, i'm new to the the language and so far things have worked pretty good and i really like the features so far.

The build system uses json for configuring how the build is going to work, that results in a bunch of errors from my editor on how the default json that the compiler generates is incorrect, following the rules of json that is.  This hints to me that maybe the json format is not the best for this objective, so that's why i'm opening this discussion.

If we like the idea of config files, maybe the toml format would be a bit better? It supports all the features we use with json but also allows for comments and the configuration format is more declarative rather than nested with json.

Or maybe we could use a c3 file, kinda like zig does. That way we could do more complicated stuff and possibly integrate better with the "c3 scripts" for building. 

I would be interested in helping with any of this if that's the case, but i want to discuss first if it makes sense.

**Comment by lerno** [comment]
It is really json5, but I am reluctant to rename the file ending, as this reduces the overall support. The intention is really for the format to be used by a program and less so by a user. It might be that I add an editor for the settings, in which case the need for comments goes away and we can forget about the comments entirely.

It used to be TOML, but I wanted a format that is more natural that programs manipulate them.

**Reply by ygorpontelo** [reply]
Is this file going to be hidden from the user in the future?

We can convert toml to json or a dict like structure, maybe presenting as toml could be a bit better for the user, but we transform it to something else when handle it in code. The current format is not bad per say, but toml seems more correct. This could be because i've dealt with toml in other languages for config so i've grown used to it.

**Reply by lerno** [reply]
It's always going to be available, but it is convenient for IDEs to manipulate it directly. That way the IDEs don't need "shadow" project settings, as is often the case.

**Reply by ygorpontelo** [reply]
Ah that's interesting, from that perspective this format is better i think. I'm also setting for json5 so the "errors" should go away. Could be interesting to mention in the docs so people expect this format. Thanks for the discussion!
